crates/ruma-client-api/src/discovery/get_supported_versions.rs

```diff
@@ -115,10 +114,12 @@ mod tests {
             "r0.5.0".to_owned(),
             "r0.0.1".to_owned(),
         ]);
-        assert_eq!(
-            sorted_reverse.known_versions().collect::<Vec<_>>(),
-            vec![MatrixVersion::V1_0, MatrixVersion::V1_1, MatrixVersion::V1_2],
-        );
+        let known_versions = sorted_reverse.known_versions();
+        assert_eq!(known_versions.len(), 3);
+        let mut iter = known_versions.into_iter();
+        assert_eq!(iter.next(), Some(MatrixVersion::V1_0));
+        assert_eq!(iter.next(), Some(MatrixVersion::V1_1));
+        assert_eq!(iter.next(), Some(MatrixVersion::V1_2));
```

Comment:
If we do this, we can do `assert_eq!(sorted_reverse.known_versions(), BTreeSet::from_iter([...]));`.

Reply:
Changed